plugins/todo/src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Table,
+  TableColumn,
+  useApi,
+  OverflowTooltip,
+  Link,
+} from '@backstage/core';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import Alert from '@material-ui/lab/Alert';
+import React, { useState } from 'react';
+import { todoApiRef } from '../../api';
+import { TodoItem, TodoListOptions } from '../../api/types';
+
+const PAGE_SIZE = 10;
+
+const columns: TableColumn<TodoItem>[] = [
+  {
+    title: 'Text',
+    field: 'text',
+    width: '100%',
+    highlight: true,
+    render: ({ text }) => <OverflowTooltip text={text} />,
+  },
+  {
+    title: 'File',
+    field: 'repoFilePath',
+    width: '80%',
+    render: ({ viewUrl, repoFilePath }) =>
+      viewUrl ? (
+        <Link to={viewUrl} target="_blank">
```

Comment:
we should probably get the internal Link behaviors into this one eventually, like adding `noopener noreferrer` etc automatically